jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -392,6 +394,17 @@ void checkAccess(Class<?> clazz) throws IllegalAccessException {
 		private void checkAccess(Class<?> targetClass, String name, int memberModifiers, MethodHandle handle) throws IllegalAccessException {
 			checkClassAccess(targetClass);
 
+			/*[IF Sidecar19-SE]*/
+			if (null != handle) {
+				MethodType type = handle.type();
+				Module accessModule = accessClass.getModule();
+				checkClassModuleVisibility(accessModule, targetClass, false);
+				checkClassModuleVisibility(accessModule, type.returnType, true);
+				for (Class<?> c: type.parameterArray()) {
```

Comment:
`type.parameterArray()` allocates a new array - since this code is in the j.l.i package and we know it won't modify the array, better to use `type.arguments` directly